torchrl/modules/models/multiagent.py

```diff
@@ -2,20 +2,113 @@
 #
 # This source code is licensed under the MIT license found in the
 # LICENSE file in the root directory of this source tree.
+from __future__ import annotations
 
+import abc
 from typing import Optional, Sequence, Tuple, Type, Union
 
 import numpy as np
 
 import torch
-from torch import nn
 
+from tensordict import TensorDict
+from torch import nn
 from torchrl.data.utils import DEVICE_TYPING
 
 from torchrl.modules.models import ConvNet, MLP
 
 
-class MultiAgentMLP(nn.Module):
+class MultiAgentNetBase(nn.Module):
+    """A base class for multi-agent networks."""
+    _empty_net: nn.Module
+
+    def __init__(
+        self,
+        *,
+        n_agents: int,
+        centralised: bool,
+        share_params: bool,
+        agent_dim: int,
+        **kwargs,
+    ):
+        super().__init__()
+
+        self.n_agents = n_agents
+        self.share_params = share_params
+        self.centralised = centralised
+        self.agent_dim = agent_dim
+
+        agent_networks = [
+            self._build_single_net(**kwargs)
+            for _ in range(self.n_agents if not self.share_params else 1)
+        ]
+        kwargs["device"] = "meta"
+        self.__dict__["_empty_net"] = self._build_single_net(**kwargs)
```

Comment:
Just curious the resason for these 2 lines. I don't understand them

Reply:
To execute the functional module we need an "empty" copy (with meta device - ie no device - ie no data). We also want that copy not to be registered within the list of modules to avoid having those empty parameters pop up when we cann `module.parameters()`.